dot/state/grandpa.go

```diff
@@ -252,12 +250,11 @@ func (s *GrandpaState) SetNextResume(number *big.Int) error {
 	return s.db.Put(resumeKey, number.Bytes())
 }
 
-// GetNextResume returns the block number of the next grandpa resume, nil if there is no upcoming resume
+// GetNextResume returns the block number of the next grandpa resume.
+// It returns an error on failure, which is contains the
+// chaindb.ErrKeyNotFound error if the key is not found.
 func (s *GrandpaState) GetNextResume() (*big.Int, error) {
 	num, err := s.db.Get(resumeKey)
-	if err == chaindb.ErrKeyNotFound {
-		return nil, nil
-	}
```

Comment:
same here

Reply:
I changed the callers to check for `chaindb.ErrKeyNotFound` and adapt their behavior.
Should I still revert this and nolint this linter?

Reply:
ok, same as above, if the caller logic checks it should be ok.